pace-util/tests/test_zarr_monitor.py

```diff
@@ -453,21 +463,57 @@ def test_transposed_diags_write_across_ranks(diag, cube_partitioner, tmpdir_fact
 
 
 @requires_zarr
-def test_transposed_diags_write_across_timesteps(
-    diag, cube_partitioner, tmpdir_factory
-):
+def test_transposed_diags_write_across_timesteps(diag, zarr_monitor_single_rank):
 
-    tmpdir = tmpdir_factory.mktemp("data.zarr")
-    store = zarr.storage.DirectoryStore(tmpdir)
-    monitor = pace.util.ZarrMonitor(store, cube_partitioner)
     # verify that we can store transposed diags across time
     time_1 = cftime.DatetimeJulian(2010, 6, 20, 6, 0, 0)
     diag_1 = _transpose(
         diag, dims_2d=[Y_DIMS, X_DIMS], dims_3d=[Z_DIMS, Y_DIMS, X_DIMS]
     )
-    monitor.store({"time": time_1, "a": diag_1})
+    zarr_monitor_single_rank.store({"time": time_1, "a": diag_1})
     time_2 = cftime.DatetimeJulian(2010, 6, 20, 6, 15, 0)
     diag_2 = _transpose(
         diag, dims_2d=[X_DIMS, Y_DIMS], dims_3d=[X_DIMS, Y_DIMS, Z_DIMS]
     )
-    monitor.store({"time": time_2, "a": diag_2})
+    zarr_monitor_single_rank.store({"time": time_2, "a": diag_2})
+
+
+@requires_zarr
+def test_diags_fail_different_dim_set(diag, numpy, zarr_monitor_single_rank):
+    time_1 = cftime.DatetimeJulian(2010, 6, 20, 6, 0, 0)
+    time_2 = cftime.DatetimeJulian(2010, 6, 20, 6, 15, 0)
+    zarr_monitor_single_rank.store({"time": time_1, "a": diag})
+    new_dims = list(diag.dims)
+    new_dims[-1] = "some_other_dim"
+    diag_2 = pace.util.Quantity(
+        numpy.ones([size + 2 for size in range(len(diag.dims))]),
+        dims=new_dims,
+        units="m",
+    )
+    with pytest.raises(ValueError):
```

Comment:
You may want to test that the correct error message is printed, if you care about your more user-friendly error message versus the one in `.transpose`.